node_modules/col-assets/lib/api.js

```diff
@@ -957,75 +958,135 @@ var canDeleteAsset = function(user, asset) {
 /* PREVIEWS */
 
 /**
- * Generate a thumbnail, large image preview and embed code for an asset
+ * Generate previews for an asset
  *
- * @param  {Asset}          asset                           The asset for which previews are generated
- * @api private
+ * @param  {Asset}        asset             The asset for which previews are generated
+ * @param  {Function}     callback          Standard callback function
+ * @param  {Object}       callback.err      An error object, if any
  */
-var generatePreviews = function(asset) {
+var generatePreviews = module.exports.generatePreviews = function(asset, callback) {
+  callback = callback || function(err) {
+    if (err) {
+      log.error({'err': err, 'asset': asset.id}, 'Failed to contact the previews service to generate a preview');
```

Comment:
Not in this function no? Generating the actual previews can fail for multiple reasons, but that's not being dealt with here